Wrappers/Python/ccpi/viewer/CILViewer.py

```diff
@@ -822,12 +831,20 @@ def saveRender(self, filename, renWin=None):
         w2if.SetInput(renWin)
         w2if.Update()
 
-        # Check if user has supplied an extension
-        extn = os.path.splitext(filename)[1]
-        if extn.lower() == '.png':
-                saveFilename = filename
-        else:
-            saveFilename = filename+'.png'
+        if not hasattr(self, 'saved_renders'):
+            self.saved_renders = 0
+
+        basename = os.path.splitext(filename)[0]
+        first_attempt = True
+        
+        while first_attempt or os.path.exists(saveFilename):
+            str_number = str(self.saved_renders)
+            while len(str_number) < 3:
+                str_number = '0' + str_number
+ 
+            saveFilename = basename + str_number + '.png'
```

Comment:
I think this can be done simply with

```python 
saveFilename = '{}_{:04d}.png'.format(basename, self.saved_renders)
```

Rather than keeping the counter, I was thinking to look in the directory and see if there were files matching the `fname_string` along these lines

```python 
import re, glob

fname_string = '{}_([0-9]*).png'.format(basename)
res_dir = os.path.dirname(filename)

for el in glob.glob(str(res_dir)):
    slist.append(int(re.search(fname_string, str(os.path.basename(el))).group(1)))
```
then `slist` contains a list of the numbers in the `fname_string`: i.e. if the string is `myfile` and in the directory there are `myfile_0000.png` and `myfile_0001.png` it should return 0 and 1. So you now know that the next number needs to be 2.